cmd/descheduler/app/descheduler_test.go

```diff
@@ -0,0 +1,128 @@
+/*
+Copyright 2024 The Karmada Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package app
+
+import (
+	"net/http"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/karmada-io/karmada/cmd/descheduler/app/options"
+)
+
+func TestNewDeschedulerCommand(t *testing.T) {
+	stopCh := make(chan struct{})
+	cmd := NewDeschedulerCommand(stopCh)
+
+	assert.NotNil(t, cmd)
+	assert.Equal(t, "karmada-descheduler", cmd.Use)
+	assert.NotEmpty(t, cmd.Long)
+}
+
+func TestDeschedulerCommandFlagParsing(t *testing.T) {
+	testCases := []struct {
+		name        string
+		args        []string
+		expectError bool
+	}{
+		{"Default flags", []string{}, false},
+		{"With custom health probe bind address", []string{"--health-probe-bind-address=127.0.0.1:8080"}, false},
+		{"With custom metrics bind address", []string{"--metrics-bind-address=127.0.0.1:8081"}, false},
+		{"With leader election enabled", []string{"--leader-elect=true"}, false},
+		{"With invalid flag", []string{"--invalid-flag=value"}, true},
+	}
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			stopCh := make(chan struct{})
+			cmd := NewDeschedulerCommand(stopCh)
+			cmd.SetArgs(tc.args)
+			err := cmd.ParseFlags(tc.args)
+			if tc.expectError {
+				assert.Error(t, err)
+			} else {
+				assert.NoError(t, err)
+			}
+		})
+	}
+}
+
+func TestServeHealthzAndMetrics(t *testing.T) {
+	healthAddress := "127.0.0.1:8082"
+	metricsAddress := "127.0.0.1:8083"
+
+	go serveHealthzAndMetrics(healthAddress, metricsAddress)
```

Comment:
Too many arguments in call to 'serveHealthzAndMetrics'

Reply:
@XiShanYongYe-Chang I made the same error. Few days a ago a commit changed the 'serveHealthzAndMetrics' and the new implementation is ```
func serveHealthzAndMetrics(healthProbeBindAddress, metricsBindAddress string) {
	if healthProbeBindAddress == metricsBindAddress {
		if healthProbeBindAddress != "0" {
			go serveCombined(healthProbeBindAddress)
		}
	} else {
		if healthProbeBindAddress != "0" {
			go serveHealthz(healthProbeBindAddress)
		}
		if metricsBindAddress != "0" {
			go serveMetrics(metricsBindAddress)
		}
	}
}
```
Your local might be behind by some commits.

Reply:
Can you rebase the latest master code and re-push it?

Reply:
Okay